### PR TITLE
feat(statistics): show dives by water-temperature band

### LINKED ISSUE #1827: Statistics: distribution of dives by water-temperature band
**Is your feature request related to a problem? Please describe.**

The Conditions statistics show water temperature by month (min, average and max) and a per-dive trend (`lib/features/statistics/data/repositories/statistics_repository.dart:1504-1540`), but not how dives are distributed across temperature ranges. A diver can't see how their diving splits between cold, temperate and warm water, or how SAC and dive time change between them. Requested on ScubaBoard.

**Describe the solution you'd like**

A bucketed chart of dives by water temperature, for example below 10, 10-18, 18-24 and above 24 °C. It should follow the diver's temperature unit, with the bands defined in °F for imperial divers rather than just relabelled from Celsius. Optionally, show average SAC and bottom time per band.

**Describe alternatives you've considered**

Reading it off the monthly min, average and max chart, which mixes location and season.

**Additional context**

#864 (closed) is about decimals on the temperature graph, not grouping.

### PR DESCRIPTION
**Pull request by ericgriffin**
## Summary

Adds a **Dives by Water Temperature** card to Statistics > Conditions: a bar chart of how many dives fall in each water-temperature band, so a diver can see how their diving splits between cold, temperate, warm and tropical water.

The bands are defined in the active diver's temperature unit, not relabelled from Celsius:

| Unit | Bands |
| --- | --- |
| Celsius | <10, 10-18, 18-24, 24+ °C (the bands proposed in the issue) |
| Fahrenheit | <50, 50-65, 65-75, 75+ °F |

Closes #1827

## Changes

- `StatisticsRepository.getDivesByWaterTempBand` (new, placed next to the other temperature queries) converts each dive's stored Celsius `water_temp` to the display unit in SQL, rounds it to the one decimal `formatTemperature` shows, and buckets it against the band edges, which are passed as bound variables (the same approach `getVisibilityDistribution` takes). Lower edges are inclusive. It goes through `_diveFilter`, so it honours the active `DiveFilterState`, the current diver and per-dive statistics exclusions (the stats-scope census passes).
- `water_temp_bands.dart` (new domain file) holds the per-unit band edges and the `WaterTempBandCount` record.
- `waterTempBandDistributionProvider` watches the temperature unit, so switching between Celsius and Fahrenheit re-bins the chart.
- The Conditions page gets the new card below the two existing temperature charts. It follows the time-at-depth chart: numeric tick labels (`<10`, `10-18`, `24+`), the unit symbol shown once on the x-axis, and a dive-count tooltip and screen-reader summary.
- Five new strings, translated into all 11 locales.

## Decisions worth a look

- **Binning uses the displayed value.** Comparing raw storage values against converted edges breaks at the boundary. A UDDF file stores 65 °F as 291.48 K, which reads back as 18.33 °C, or 64.994 °F. The app shows that dive as "65°F", so it should count in 65-75, not 50-65. Rounding to display precision first makes the band agree with the number the diver sees. A test pins this, and removing the `ROUND` makes that test fail.
- **Fahrenheit edges are 50 / 65 / 75.** They are round Fahrenheit numbers close to the Celsius edges (10 °C is exactly 50 °F; 18 and 24 °C are 64.4 and 75.2 °F), so both units describe the same cold, temperate, warm and tropical split.
- **Empty bands are kept.** All four bars always show, so "I never dive below 10 °C" is visible and charts stay comparable across filters. Dives without a water temperature are left out, and the card shows an empty state when no dive in scope has one.
- **Average SAC and bottom time per band were left out** (the issue marks them optional). The existing bar-chart card holds one count per category and has no slot for per-band metrics. SAC also depends on the selected SAC/RMV lane and gas model, so bolting it on would mean a new card layout. That is better as its own follow-up than stretched into this change.

## Test Plan

- [x] New repository tests (`water_temp_band_distribution_test.dart`): Celsius bucketing with inclusive lower edges, empty bands kept, Fahrenheit re-binning that differs from Celsius, the 18.33 °C to "65°F" boundary case, tag filter, stats exclusion, diver scoping, and the empty result.
- [x] New provider test: changing the unit setting re-bins the chart (fails if the unit is read rather than watched).
- [x] New widget tests: Celsius and Fahrenheit tick labels and axis symbol, and the empty state.
- [x] Added the new method and provider to the existing repository error-handling and all-providers smoke tests.
- [x] Existing Conditions page tests, the statistics filter tests, the `DiveStatsScope` census and the ARB parity and localization tests pass.
- [x] `dart format .` and `flutter analyze` over the whole project are clean.
- [x] Pre-push hook passed.
- [ ] Manual testing: not run on a device in this change. Behaviour is covered by the repository, provider and widget tests above.

### CODE REVIEW

File: lib/features/statistics/data/repositories/statistics_repository.dart [hunk 1]
```diff
@@ -1542,6 +1544,81 @@ class StatisticsRepository {
...
+        TemperatureUnit.fahrenheit => 'water_temp * 9.0 / 5.0 + 32.0',
+      };
+      // Highest edge first so the first matching WHEN is the dive's band.
+      // Edge variables come first: they appear before the diver and filter
+      // placeholders in the statement below, and Drift binds positionally.
+      final whens = [
+        for (var i = edges.length - 1; i >= 0; i--)
+          'WHEN temp >= ? THEN ${i + 1}',
+      ].join('\n            ');
+      final params = [...edges.reversed, ?diverId, ...df.params];
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/statistics/data/repositories/statistics_repository.dart:1616]: No change: `?diverId` is a Dart 3.8 null-aware collection element (the project pins sdk ^3.10.0), so it adds nothing when `diverId` is null and the params line up with the placeholders. A scratch probe gives `[24, 18, 10, ?diverId, tag]` length 4 with a null id and 5 with one. Every repository test except the diver one runs with a null `diverId`, including the tag-filter test that also binds filter params, and all pass. The analyzer lint `use_null_aware_elements` is what rewrote the original `if (diverId != null) diverId` into this form.

File: lib/l10n/arb/app_en.arb [hunk 2]
```diff
@@ -23369,6 +23369,20 @@
   "statistics_conditions_tempTrend_subtitle": "Every dive in range",
   "statistics_conditions_tempTrend_empty": "No temperature data available",
   "statistics_conditions_tempTrend_error": "Failed to load temperature trend",
+  "statistics_conditions_waterTempBands_title": "Dives by Water Temperature",
+  "statistics_conditions_waterTempBands_subtitle": "How your dives split across water temperature bands",
+  "statistics_conditions_waterTempBands_empty": "No water temperature data available",
+  "statistics_conditions_waterTempBands_error": "Failed to load water temperature bands",
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/l10n/arb/app_en.arb:23404]: Added `@` descriptions for the title, subtitle, empty and error keys in app_en.arb and regenerated, so the generated interface now documents each string (f8661a7ad9a).
